lib/orderbook/OrderBook.ts

```diff
@@ -118,26 +118,42 @@ class OrderBook extends EventEmitter {
     return outgoingOrder ;
   }
 
+  /**
+   * Checks that a currency advertised by a peer are known to us, have a swap client identifier,
+   * and that their token identifier matches ours.
+   */
+  private isPeerCurrencySupported = (peer: Peer, currency: string) => {
+    const currencyAttributes = this.getCurrencyAttributes(currency);
+    if (!currencyAttributes) {
+      return false; // we don't know about this currency
+    }
+
+    if (!peer.getIdentifier(currencyAttributes.swapClient, currency)) {
+      return false; // peer did not provide a swap client identifier for this currency
+    }
+
+    // ensure that our token identifiers match
+    const ourTokenIdentifier = this.pool.getTokenIdentifier(currency);
+    const peerTokenIdentifier = peer.getTokenIdentifier(currency);
+    return ourTokenIdentifier === peerTokenIdentifier;
+  }
+
   private bindPool = () => {
     this.pool.on('packet.order', this.addPeerOrder);
     this.pool.on('packet.orderInvalidation', this.handleOrderInvalidation);
     this.pool.on('packet.getOrders', this.sendOrders);
     this.pool.on('packet.swapRequest', this.handleSwapRequest);
     this.pool.on('peer.close', this.removePeerOrders);
     this.pool.on('peer.pairDropped', this.removePeerPair);
-    this.pool.on('peer.pairsAdvertised', this.verifyPeerPairs);
+    this.pool.on('peer.verifyPairs', this.verifyPeerPairs);
     this.pool.on('peer.nodeStateUpdate', (peer) => {
```

Comment:
Rename `verifyPeerPairs` you mean? Yeah I'm not in love with the current naming, but maybe I'd call it `checkAdvertisedPairs`. It's technically checking both pairs and currencies, but I think the end purpose is to determine which trading pairs to activate.

Reply:
I meant moving the callback of `peer.nodeStateUpdate` to a named function (instead of having it anon. inline) like all the other listeners for pool currently are.

Reply:
Ohhh gotcha, I'll do that in a separate PR/commit.

Reply:
Sounds good.